enterprise_access/apps/subsidy_access_policy/models.py

```diff
@@ -0,0 +1,192 @@
+from uuid import uuid4
+
+from django.db import models, transaction
+from django_extensions.db.models import TimeStampedModel
+from simple_history.models import HistoricalRecords
+from enterprise_access.apps.subsidy_access_policy.constants import AccessMethods
+from enterprise_access.apps.subsidy_access_policy.mocks import group_client, subsidy_client, catalog_client
+
+
+class SubsidyAccessPolicy(TimeStampedModel):
+    """
+    Tie together information used to control access to a subsidy.
+    This abstract model joins group, catalog, and access method.
+
+    .. no_pii:
+    """
+    class Meta:
+        """
+        Metaclass for SubsidyAccessPolicy.
+        """
+        abstract = True
+
+    uuid = models.UUIDField(
+        primary_key=True,
+        default=uuid4,
+        editable=False,
+        unique=True,
+    )
+    version = models.IntegerField(default=0)
+    active = models.BooleanField(default=False)
+    group_uuid = models.UUIDField(null=True, blank=True, db_index=True)
+    catalog_uuid = models.UUIDField(null=True, blank=True, db_index=True)
+    subsidy_uuid = models.UUIDField(null=False, blank=False, db_index=True)
+    access_method = AccessMethods.DIRECT
+    history = HistoricalRecords()
+    
+    @property
+    def enterprise_customer_uuid(self):
+        "Returns the enterprise_customer_uuid. Equivalent to the group uuid at present"
+        return self.group_uuid
+
+    @abstractmethod
+    def can_redeem(self, learner_id, content_key):
+        """
+        Check that a given learner can redeem the given content.
+        """
+        if not catalog_client.catalog_contains_content(self.catalog_uuid, content_key):
+            return False
+        if not group_client.group_contains_learner(self.group_uuid, learner_id):
+            return False
+        if not subsidy_client.is_redeemable(self.subsidy_uuid, learner_id, content_key):
+            return False
+
+        return True
+
+    def redeem(self, learner_id, content_key):
+        """
+        Redeem a subsidy for the given learner and content.
+        Returns:
+            A ledger transaction id, or None if the subsidy was not redeemed.
+        """
+        if self.can_redeem(learner_id, content_key):
+            if self.access_method == AccessMethods.DIRECT:
+                return subsidy_client.redeem(self.subsidy_uuid, learner_id, content_key)
+            if self.access_method == AccessMethods.REQUEST:
+                return subsidy_client.request_redemption(self.subsidy_uuid, learner_id, content_key)
+        return None
+
+    def has_redeemed(self, learner_id, content_key):
+        """
+        Check if the subsidy has been redeemed.
+        """
+        if self.access_method == AccessMethods.DIRECT:
+            return subsidy_client.has_redeemed(self.subsidy_uuid, learner_id, content_key)
+        elif self.access_method == AccessMethods.REQUEST:
+            return subsidy_client.has_requested(self.subsidy_uuid, learner_id, content_key)
```

Comment:
Note: the prototype code is probably wrong about this branch.  If the access method is browse & request, it should really query the enterprise-access B&R API.  We may need to update the models/API for B&R to accept a subsidy uuid as a filter condition.